src/shared/components/AssetImage/AssetImage.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactElement, useEffect, cloneElement, useState, useMemo } from 'react'
+
+import { AllChannelFieldsFragment, AssetAvailability, VideoFieldsFragment } from '@/api/queries'
+import { useStorageProviders } from '@/hooks'
+import { createStorageNodeUrl } from '@/utils/asset'
+
+type AssetImageProps = { entity?: AllChannelFieldsFragment | VideoFieldsFragment | null; component: ReactElement }
+
+const AssetImage = ({ entity, component }: AssetImageProps) => {
+  const [assetUrl, setAssetUrl] = useState<string | undefined>(undefined)
+  const { getStorageProvider } = useStorageProviders()
+  const assetData = useMemo(() => {
+    if (entity && entity.__typename === 'Channel') {
+      return {
+        title: entity.title,
+        availability: entity.coverPhotoAvailability || entity.avatarPhotoAvailability,
+        urls: entity.avatarPhotoUrls || entity.coverPhotoUrls,
+        dataObject: entity.coverPhotoDataObject || entity.avatarPhotoDataObject,
```

Comment:
`coverPhoto` and `avatarPhoto` are two separate assets so we need to somehow switch which one we use